src/utils/objectUtils.ts

```diff
@@ -94,6 +96,30 @@ export function groupBy<
   )
 }
 
+/**
+ * @param array The array of objects to be grouped.
+ * @param selector The key used for grouping the objects. Support nested keys.
+ * @returns An object where the keys are unique values from the given selector and the values are the corresponding objects from the array.
+ */
+export function groupByPath<T extends object>(array: T[], selector: string): Record<string, T[]> {
```

Comment:
🟢 Could we benefit from a shorthand groupByPathUnique? I am not tbh, just asking in case it can be useful